### PR TITLE
Upgrade tap to v18

### PR DESCRIPTION
**Pull request by tarunrajput**
/claim #578

### CODE REVIEW

File: packages/plugin-parsedoc/test/config/tap.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 ---
 jobs: 5
 timeout: 120
-reporter: spec
-coverage: false
+reporter: base
+disable-coverage: true
```
Comment (micheleriva) [inline packages/plugin-parsedoc/test/config/tap.yml:5]: Coverage should be enabled

File: packages/plugin-telemetry/src/tests/config/tap.yml [hunk 2]
```diff
@@ -1,8 +1,8 @@
 ---
 jobs: 5
 timeout: 120
-reporter: spec
-coverage: false
+reporter: base
+disable-coverage: true
```
Comment (micheleriva) [inline packages/plugin-telemetry/src/tests/config/tap.yml:5]: Coverage should be enabled

File: packages/plugin-data-persistence/test/config/tap.yml [hunk 3]
```diff
@@ -1,8 +1,8 @@
 ---
 jobs: 5
 timeout: 120
-reporter: spec
-coverage: false
+reporter: base
+disable-coverage: true
```
Comment (micheleriva) [inline packages/plugin-data-persistence/test/config/tap.yml:5]: Coverage should be enabled

File: packages/orama/tests/config/tap.yml [hunk 4]
```diff
@@ -1,8 +1,8 @@
 ---
 jobs: 5
 timeout: 120
-reporter: spec
-coverage: false
+reporter: base
+disable-coverage: true
```
Comment (micheleriva) [inline packages/orama/tests/config/tap.yml:5]: Coverage should be enabled

File: packages/plugin-match-highlight/test/config/tap.yml [hunk 5]
```diff
@@ -1,8 +1,8 @@
 ---
 jobs: 5
 timeout: 120
-reporter: spec
-coverage: false
+reporter: base
+disable-coverage: true
```
[the remark this answers is not in the document]
Comment (tarunrajput) [reply packages/plugin-match-highlight/test/config/tap.yml:5]: @micheleriva, aren't we using c8 for that?

[the remark this answers is not in the document]
Comment (micheleriva) [reply packages/plugin-match-highlight/test/config/tap.yml:5]: tests doesn't pass so I couldn't verify :(

**Comment by tarunrajput** [comment]
@micheleriva, some tests are failing in concurrent mode when running `turbo test`, but they execute without issues when using `turbo test --concurrency=1.` Do you have any idea into what might be causing this?

File: packages/plugin-match-highlight/test/config/tap.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (tarunrajput) [reply packages/plugin-match-highlight/test/config/tap.yml:5]: **with coverage disabled:**

image

---
**with coverage enabled:**

image

[the remark this answers is not in the document]
Comment (micheleriva) [reply packages/plugin-match-highlight/test/config/tap.yml:5]: alright looks good!

**Comment by micheleriva** [comment]
@tarunrajput if you merge main into your branch, it should bring the fix for the tests

**Comment by micheleriva** [comment]
Hi @tarunrajput, any update on this?

**Comment by tarunrajput** [comment]
@micheleriva , I'm having a strange issue. tests pass when running pnpm test for individual packages, but they fail when using turbo.

**Comment by micheleriva** [comment]
@tarunrajput have you been able to merge main into your branch? We fixed this

**Comment by tarunrajput** [comment]
@micheleriva, getting same issue after rebasing with main :(

**Review by micheleriva — APPROVED** [review]
LGTM

**Pull request merged**